src/realm/sync/subscriptions.cpp

```diff
@@ -57,6 +57,78 @@ constexpr static std::string_view c_flx_sub_query_str_field("query");
 
 using OptionalString = util::Optional<std::string>;
 
+enum class SubscriptionStateForStorage : int64_t {
+    // The subscription set has been persisted locally but has not been acknowledged by the server yet.
+    Pending = 1,
+    // The server is currently sending the initial state that represents this subscription set to the client.
+    Bootstrapping = 2,
+    // This subscription set is the active subscription set that is currently being synchronized with the server.
+    Complete = 3,
+    // An error occurred while processing this subscription set on the server. Check error_str() for details.
+    Error = 4,
+    // The last bootstrap message containing the initial state for this subscription set has been received. The
+    // client is awaiting a mark message to mark this subscription as fully caught up to history.
+    AwaitingMark = 6,
+};
+
+SubscriptionSet::State state_from_storage(int64_t value)
+{
+    switch (static_cast<SubscriptionStateForStorage>(value)) {
+        case SubscriptionStateForStorage::Pending:
+            return SubscriptionSet::State::Pending;
+        case SubscriptionStateForStorage::Bootstrapping:
+            return SubscriptionSet::State::Bootstrapping;
+        case SubscriptionStateForStorage::AwaitingMark:
+            return SubscriptionSet::State::AwaitingMark;
+        case SubscriptionStateForStorage::Complete:
+            return SubscriptionSet::State::Complete;
+        case SubscriptionStateForStorage::Error:
+            return SubscriptionSet::State::Error;
+        default:
+            throw std::runtime_error(util::format("Invalid state for SubscriptionSet stored on disk: %1", value));
+    }
+}
+
+int64_t state_to_storage(SubscriptionSet::State state)
+{
+    switch (state) {
+        case SubscriptionSet::State::Pending:
+            return static_cast<int64_t>(SubscriptionStateForStorage::Pending);
+        case SubscriptionSet::State::Bootstrapping:
+            return static_cast<int64_t>(SubscriptionStateForStorage::Bootstrapping);
+        case SubscriptionSet::State::AwaitingMark:
+            return static_cast<int64_t>(SubscriptionStateForStorage::AwaitingMark);
+        case SubscriptionSet::State::Complete:
+            return static_cast<int64_t>(SubscriptionStateForStorage::Complete);
+        case SubscriptionSet::State::Error:
+            return static_cast<int64_t>(SubscriptionStateForStorage::Error);
+        default:
+            REALM_UNREACHABLE();
+    }
+}
+
+size_t state_to_order(SubscriptionSet::State needle)
+{
+    using State = SubscriptionSet::State;
+    switch (needle) {
+        case State::Uncommitted:
+            return 0;
+        case State::Pending:
+            return 1;
+        case State::Bootstrapping:
+            return 2;
+        case State::AwaitingMark:
```

Comment:
Shouldn't this be the last one and return 6 instead?

Reply:
Yeah, that's so we don't break the file format.